src/Filter/StringFilter.php

```diff
@@ -68,10 +70,18 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
                 $this->getOption('case_sensitive') ? $data['value'] : mb_strtolower($data['value'])
             );
         } else {
+            if (ContainsOperatorType::TYPE_STARTS_WITH === $data['type']) {
+                $format = '%s%%';
+            } elseif (ContainsOperatorType::TYPE_ENDS_WITH === $data['type']) {
+                $format = '%%%s';
+            } else {
+                $format = $this->getOption('format');
+            }
```

Comment:
You can use a switch on $data['type'].

@sonata-project/contributors I feel like the `format` option should be deprecated. WDYT ?